fire/SFMainMod.F90

```diff
@@ -454,6 +477,36 @@ real(r8) a,b,c,e              ! function of fuel sav
        ! remove mineral content from net fuel load per Thonicke 2010 for ir calculation
        currentPatch%sum_fuel  = currentPatch%sum_fuel * (1.0_r8 - SF_val_miner_total) !net of minerals
 
+       ! ---------------------------------------------------
+       ! Active crown fire effects: https://github.com/NGEET/fates/issues/573
+       ! Update some characteristics of fuel
+       sum_fuel = currentPatch%sum_fuel  ! save for comparison later
+       if (currentPatch%fire == 1) then
+          currentCohort=>currentPatch%tallest
+          do while(associated(currentCohort))
+             if (currentCohort%active_crown_fire_flg == 1) then
+                ! Add the leaf carbon from each cohort to currentPatch%sum_fuel
+                leaf_c = currentCohort%prt%GetState(leaf_organ, all_carbon_elements)
+                currentPatch%sum_fuel = currentPatch%sum_fuel + leaf_c
+             end if
+             currentCohort => currentCohort%shorter;
+          enddo !end cohort loop
+
+          ! if sum_fuel was indeed updated for a case of active crown fire, go
+          ! on to update currentPatch%fuel_mef and currentPatch%fuel_eff_moist
+          if (currentPatch%sum_fuel > sum_fuel) then
+             alpha_live_fuel = (currentPatch%livegrass + leaf_c) /  &
+                               (currentPatch%livegrass + leaf_c + sum(litt_c%leaf_fines(:)) + litt_c%ag_cwd(tw_sf))
+             fuel_eff_moist_dead = sum(currentPatch%fuel_frac(tw_sf:dl_sf) * fuel_moisture(tw_sf:dl_sf))
+             fuel_mef_fine = currentPatch%fuel_frac(tw_sf) * MEF(tw_sf) +  &
+                             currentPatch%fuel_frac(dl_sf) * MEF(dl_sf) +  &
+                             currentPatch%fuel_frac(lg_sf) * MEF(lg_sf)
+             currentPatch%fuel_mef = max((2.9_r8 * ((1.0_r8 - alpha_live_fuel) / alpha_live_fuel) * (1.0_r8 - fuel_eff_moist_dead) - 0.226_r8), fuel_mef_fine)
+             currentPatch%fuel_eff_moist = exp(-1.0_r8 * currentSite%acc_NI * SF_val_SAV(lg_sf) / SF_val_drying_ratio)
+          end if
+       end if
+       ! ---------------------------------------------------
```

Comment:
@jkshuman I am interested in your comments throughout, but I'm least confident about this section. Pls let me know if I interpreted [your recommendations](https://github.com/NGEET/fates/issues/573#issuecomment-528641079) correctly.

Reply:
To clarify my request for comments here:
- Most important right now is whether I got the formulas right.
- I recognize that we may consider moving this piece of code to subr. characteristics_of_fuel. If so, we would need to call subr. characteristics_of_fuel a second time, just before calling rate_of_spread its second time.

Reply:
Oh, I did not use this suggested formula:
`fuel_sav_crown =  tree_lai/canopy_depth`
because the SF_val_SAV values in the fates params file suggest to me that this is an area-to-volume that should not include the canopy air-space. Pls correct me if I'm mistaken.

Reply:
> [...] we may consider moving this piece of code to subr. characteristics_of_fuel. If so, we would need to call subr. characteristics_of_fuel a second time, just before calling rate_of_spread its second time.

And if so, we will need to replace this line (which I moved to the end of subr. characteristics_of_fuel):
`currentPatch%sum_fuel  = currentPatch%sum_fuel * (1.0_r8 - SF_val_miner_total) !net of minerals`
with
`currentPatch%sum_fuel_minus_minerals  = currentPatch%sum_fuel * (1.0_r8 - SF_val_miner_total) !net of minerals`
to avoid removing the minerals twice.

Reply:
Following our 11/1 conversation, I have now confirmed that `currentPatch%sum_fuel` does not get updated during the ground-fire phase. This means that it obtains its value in `subroutine characteristics_of_fuel` and maintains that value until the second call to `subroutine rate_of_spread` when crown leaves get added to it. This is what we want.